### PR TITLE
doc: net: context: Add comment to use BSD socket API instead

### PR DESCRIPTION
**Pull request by jukkar**
The net_context API is being phased out.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

### CODE REVIEW

File: doc/reference/networking/net_context.rst [hunk 1]
```diff
@@ -6,6 +6,8 @@ Networking Context
 Overview
 ********
 
+This API is being phased out and user should use :ref:`bsd_sockets_interface`
```
Comment (tbursztyka) [inline doc/reference/networking/net_context.rst]: it will be still here, but only for the core part, so maybe rephrase "phased out"?

Comment (jukkar) [reply doc/reference/networking/net_context.rst]: Yes, I can modify it to say that it is not meant for application use any more.

File: doc/reference/networking/net_context.rst [hunk 2]
```diff
@@ -6,6 +6,8 @@ Networking Context
 Overview
 ********
 
+This API is not meant for application use. Application should use
```
Comment (dbkinder) [inline doc/reference/networking/net_context.rst]:
```suggestion
This API is deprecated. Applications should use
```

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: doc/reference/networking/net_context.rst (hunk 2 above)
Comment (jukkar) [reply doc/reference/networking/net_context.rst]: Actually we are not deprecating it as it will be used internally in the stack. Here I am understanding that the "deprecating" word means that it is to be removed from the system, correct if I am wrong here. We are trying to push people to use the BSD socket API instead of net_context one.

Comment (dbkinder) [reply doc/reference/networking/net_context.rst]: Ah, in that case, maybe we should delete the API documentation below (lines 12-EOF)?

Comment (jukkar) [reply doc/reference/networking/net_context.rst]: Sure, that makes sense.

**Comment by jukkar** [comment]
Removed the API reference chapter.

**Review by tbursztyka — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]
Thanks..

**Review by nashif — APPROVED** [review]
